### PR TITLE
Allow to override the path to save profile dump 

### PR DESCRIPTION
**Pull request by andypost**
- removes useless dependency
- it will not fail if extra source for xhprof will be missing
- allows to override directory where to save the profile dump

### CODE REVIEW

File: src/Commands/core/XhprofCommands.php [hunk 1]
```diff
@@ -136,14 +136,17 @@ public function xhprofFinishRun($namespace)
...
-            $xhprof_runs = new \XHProfRuns_Default();
-            return $xhprof_runs->save_run($xhprof_data, $namespace);
+            $data = \xhprof_disable();
+            if (class_exists('\XHProfRuns_Default')) {
+                $xhprof_runs = new \XHProfRuns_Default($this->getConfig()->get('xh.path'));
+                return $xhprof_runs->save_run($data, $namespace);
+            }
         }
+        $config = $this->getConfig();
+        $dir = $config->get('xh.path', $config->tmp());
```
Comment (andypost) [inline src/Commands/core/XhprofCommands.php:147]: Maybe better to use `xh.output-dir` for option?

Comment (andypost) [reply src/Commands/core/XhprofCommands.php:147]: less to type sounds better

**Comment by weitzman** [comment]
IMO _lox/xhprof_ is not a useless dependency. With it, sites can use XHProf UI without any more software. Removing it is a DX regression. One fewer dependency is always nice, though I note that this one has never caused a single bug report.

> it will not fail if extra source for xhprof will be missing

Can you elaborate?

> allows to override directory where to save the profile dump

Drush is currently out of the game if caring about output path. We delegate that to lox/xhprof so I dont love adding this.

**Comment by andypost** [comment]
@weitzman this is require-dev dependency so it never installed except drush tests

**Comment by weitzman** [comment]
Oh, that would explain the lack of bug reports :)

**Comment by andypost** [comment]
> never caused a single bug report.

Because it never used to work( 

Moreover the only working profiler for php8 is tideways which has to put everything into tmp, and xhprof has the setting only for scripts https://github.com/longxinH/xhprof/blob/master/extension/xhprof.c#L110

The real problem with lox is it's not maintained (outdated for 5 years) over xhprof from pecl https://github.com/lox/xhprof

**Pull request merged**